server/server.py

```diff
@@ -180,6 +181,78 @@ def count_verified():
     return jsonify({'verified_count': count})
 
 
+@app.route('/submit/testimonial', methods=['POST'])
+@schema.validate({
+    'properties': {
+        'name': {'type': 'string'},
+        'student_number': {'type': 'number'},
+        'program': {'type': 'string'},
+        'year': {'type': 'string'},
+        'testimonial': {'type': 'string', 'maxLength': 1500},
+        'anonymous': {'type': 'boolean'}
+    },
+    'required': ['name', 'student_number', 'program', 'year', 'testimonial']
+})
+def submit_testimonial():
+    '''
+    Submit a testimonial to be approved by the team
+    '''
+    logger.info('Hit route %s', request.path)
+
+    data = request.get_json()
+
+    logger.debug(data)
+
+    bytes_student_number = int_to_bytes(data['student_number'])
+    bytes_student_number = data['name'].encode('utf-8')
+
+    name = ''
+
+    if data['anonymous'] is True:
+        name = 'Anonymous'
+    else:
+        name = data['name'].capitalize().split(' ')[0]
+
+    testimonial = {
+        'encrypted_name': crypto.encrypt(bytes_student_number),
+        'student_number': crypto.encrypt(bytes_student_number),
+        'name': name,
+        'program': data['program'],
+        'year': data['year'],
+        'testimonial': data['testimonial'],
+        'verified': False
+    }
+
+    _id = db.add_testimonial(testimonial)
+
+    if _id is None:
+        error = jsonify({'error': 'Could not submit testimonial', 'success': False})
+        return make_response(error, 400)

```

Comment:
I believe `make_response` will `jsonify` a `dict` for you, but you may want to double check it: https://flask.palletsprojects.com/en/1.1.x/api/#flask.Flask.make_response

Reply:
I'm still cool with jsonifying separately since these are long enough that I don't think I'd put it in-line anyhow